lib/API/DX/Device.cpp

```diff
@@ -1169,18 +1169,19 @@ class DXDevice : public offloadtest::Device {
 };
 } // namespace
 
-llvm::Error Device::initializeDXDevices() {
+llvm::Error Device::initializeDXDevices(const DeviceConfig Config) {
 #ifdef _WIN32
-#ifndef NDEBUG
-  ComPtr<ID3D12Debug1> Debug1;
+  if (Config.EnableDebugLayer || Config.EnableValidationLayer) {
+    ComPtr<ID3D12Debug1> Debug1;
 
-  if (auto Err = HR::toError(D3D12GetDebugInterface(IID_PPV_ARGS(&Debug1)),
-                             "failed to create D3D12 Debug Interface"))
-    return Err;
+    if (auto Err = HR::toError(D3D12GetDebugInterface(IID_PPV_ARGS(&Debug1)),
+                               "failed to create D3D12 Debug Interface"))
+      return Err;
 
-  Debug1->EnableDebugLayer();
-  Debug1->SetEnableGPUBasedValidation(true);
-#endif
+    if (Config.EnableDebugLayer)
+      Debug1->EnableDebugLayer();
```

Comment:
Is it intentional that just setting EnableValidationLayer wouldn't actually enable the validation layer.  Both flags would need to be set.

Reply:
Does GBV require the debug layer? I assumed they were separate runtime settings.

Reply:
I think the docs indicate that you do need the debug layer for GBV, so I've updated the code.